http_handler_echo.h

```diff
@@ -14,6 +14,7 @@ struct request;
 class handler_echo : public handler {
 public:
 
+    handler_echo(std::string base_url) : handler(base_url) {}; 
```

Comment:
Similar to a previous comment, and also needs a newline to separate this line from the below comment.

`handler_echo(const std::string& root_url) : handler(root_url) {}`

I've also been enforcing comments above every method.
The professor in class said not to do this, but I disagree strongly.
Sure - for a really basic constructor like this it doesn't add any informational value,
but it sure makes the code easier to look at when there is a comment above every method instead of just most.